collection-scripts/gather_bm_network_logs

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+. gather_service_logs_util
+BASE_COLLECTION_PATH="must-gather"
+NETWORK_LOG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/bm_network_logs"}
+SERVICE_LOG_PATH="${NETWORK_LOG_PATH}"
+
+mkdir -p "${NETWORK_LOG_PATH}"/
+
+function gather_bm_network_logs {
+  POD_INFO=($(oc get pods -A -o=custom-columns=name:.metadata.name,ns:metadata.namespace| grep "${POD_NAME}"))
+  MG_NAMESPACE=${POD_INFO[1]}
+  if [ "${MG_NAMESPACE}" == "ns" ]; then
+    # POD_NAME env variable is passed to must-gather pods since 4.10 version,
+    # if is not defined, command will return the first line of 'get pods' output, which is column name='ns'
+    echo "ERROR: Collection failed, use the latest oc version for it to succeed"
+    exit 1
+  fi
+
+  # Run debug pod on every node first to reduce 'oc wait' time
+  HOST_PODS=()
+  for NODE in ${NODES[@]}; do
+    mkdir -p "${NETWORK_LOG_PATH}/${NODE}"
+    HOST_POD=$(oc debug --to-namespace="${MG_NAMESPACE}" node/"${NODE}" -o jsonpath='{.metadata.name}')
+    oc debug --to-namespace="${MG_NAMESPACE}" node/"${NODE}" -- /bin/bash -c \
+      'trap : TERM INT; sleep infinity & wait' > /dev/null 2>&1 &
+    HOST_PODS+=($HOST_POD)
+  done
+
+  # wait for pod creation
+  sleep 1
+
+  for (( i=0;i<${#NODES[@]};i++ )); do
+    NODE=${NODES[i]}
+    HOST_POD=${HOST_PODS[i]}
+    oc wait -n="${MG_NAMESPACE}" --for=condition=Ready pod/"${HOST_POD}" --timeout=30s
+    if [ $? -ne 0 ]; then
+      echo "ERROR: Timeout waiting for pod to become Ready"
+      oc delete pod -n="${MG_NAMESPACE}" "${HOST_POD}" & PIDS+=($!)
+      continue
+    fi
+
+    oc exec -n="${MG_NAMESPACE}" "${HOST_POD}" -- bash -c "ip addr" > "${NETWORK_LOG_PATH}/${NODE}/ip-addr.txt"
+    oc exec -n="${MG_NAMESPACE}" "${HOST_POD}" -- bash -c "ip route" > "${NETWORK_LOG_PATH}/${NODE}/ip-route.txt"
+    oc exec -n="${MG_NAMESPACE}" "${HOST_POD}" -- bash -c "hostname" > "${NETWORK_LOG_PATH}/${NODE}/hostname.txt"
+    HOST_COMMAND='for c in $(nmcli -f uuid con show | tail -n +2); do nmcli con show $c; done'
+    oc exec -n="${MG_NAMESPACE}" "${HOST_POD}" -- bash -c \
+      "chroot /host /bin/bash -c '${HOST_COMMAND}'" \ > "${NETWORK_LOG_PATH}/${NODE}/nmcli.txt"
+
+    oc cp -n="${MG_NAMESPACE}" "${HOST_POD}":/host/etc/resolv.conf "${NETWORK_LOG_PATH}/${NODE}/resolv.conf"
+
+    oc delete pod -n="${MG_NAMESPACE}" "${HOST_POD}" & PIDS+=($!)
+    collect_service_logs "${NODE}" NetworkManager
+  done
+}
+
+PIDS=()
+NODES=(${@:-$(oc get nodes -o jsonpath='{.items[?(@.status.nodeInfo.operatingSystem=="linux")].metadata.name}')})
+
+echo "WARNING: Collecting network logs on ALL linux nodes in your cluster. This could take a long time."
+gather_bm_network_logs
+echo "INFO: BM network log collection complete."
+wait "${PIDS[@]}"
```

Comment:
We don't take advantage of this and I think, that's an issue (for speed/threading). 

gather_bm_network_logs should be a function that is node specific (IMO) and you should subprocess it's invocation and background it so that you can thread the collection to multiple nodes at once vs it being done in serial (the way it is happening now). 

To do this it will take a bit of rework but the changes are likely small/manageable. Can this be considered?

Reply:
Absolutely!